### PR TITLE
allow image updates to be forced if container not present #181

### PR DESCRIPTION
**Pull request by iamnoah**
Supports #187. I've tested against my deployment of a PodTemplate it works wonderfully :)

### CODE REVIEW

File: pkg/argocd/argocd.go [hunk 1]
```diff
@@ -187,6 +187,10 @@ func FilterApplicationsForUpdate(apps []v1alpha1.Application, patterns []string)
 	return appsForUpdate, nil
 }
 
+func parseImageList(updateImage string) []string {
+	return strings.Split(updateImage, ",")
+}
```
Comment (jannfis) [inline pkg/argocd/argocd.go:192]: I like that this becomes externalized in its own method. Maybe it would make sense to also pull in the white-space trimming into this method, what do you think? Also, I think it could be worth to spend this method its own unit test :)

File: pkg/argocd/argocd.go [hunk 2]
```diff
@@ -417,6 +421,18 @@ func GetImagesFromApplication(app *v1alpha1.Application) image.ContainerImageLis
 		images = append(images, image)
 	}
 
+	annotations := app.Annotations
+	if updateImage, ok := annotations[common.ImageUpdaterAnnotation]; ok {
```
Comment (jannfis) [inline pkg/argocd/argocd.go:425]: Could you please add some explaining commentary to this block? It's easy to understand in the context of this pull request, but I think it might be worth to add some inline documentation explaining why we're doing this so people without the context will understand later on.

File: pkg/argocd/argocd_test.go [hunk 3]
```diff
@@ -56,6 +56,26 @@ func Test_GetImagesFromApplication(t *testing.T) {
 		imageList := GetImagesFromApplication(application)
 		assert.Empty(t, imageList)
 	})
+
+	t.Run("Get list of images from application that has force-update", func(t *testing.T) {
```
Comment (jannfis) [inline pkg/argocd/argocd_test.go:60]: Tests are always appreciated, thank you! :+1:

File: docs/configuration/images.md [hunk 4]
```diff
@@ -61,6 +61,21 @@ of the [Semver library](https://github.com/Masterminds/semver) we're using.
     [filtering tags](#filtering-tags)
     below.
 
+
+### Forcing Image Updates
```
Comment (jannfis) [inline docs/configuration/images.md:65]: +1 for documentation update :)

**Review by jannfis — COMMENTED** [review]
Thank you @iamnoah for taking the time and effort to submit this new feature!

Your PR looks awesome, and I have no formal request for change. Just some items up for discussion - please see below.

File: pkg/argocd/argocd.go (hunk 1 above)
Comment (iamnoah) [reply pkg/argocd/argocd.go]: Once I pulled in the white space, I noticed the next step was to convert to a ContainerImage so I pulled that in as well. I just focused tests on the string parsing since NewFromIdentifier seems well covered.

File: pkg/argocd/argocd_test.go [hunk 5]
```diff
@@ -702,3 +722,10 @@ func TestKubernetesClient_UpdateSpec_Conflict(t *testing.T) {
 
 	assert.Equal(t, "https://github.com/argoproj/argocd-example-apps", spec.Source.RepoURL)
 }
+
+func Test_parseImageList(t *testing.T) {
+	assert.Equal(t, []string{"foo", "bar"}, parseImageList(" foo, bar ").Originals())
+	// should whitespace inside the spec be preserved?
```
Comment (jannfis) [inline pkg/argocd/argocd_test.go:728]: I think whitespace stripping should be done here, but let's do that in another PR: https://github.com/argoproj-labs/argocd-image-updater/blob/f1cc8e4542b0531f66fc5546f7467833b1d08ebe/pkg/image/image.go#L133

**Review by jannfis — APPROVED** [review]
LGTM, thank you @iamnoah !
